lib/spack/spack/test/mirror.py

```diff
@@ -107,6 +109,59 @@ def test_url_mirror(mock_archive):
     repos.clear()
 
 
+def test_bad_mirror_args():
```

Comment:
Same request for having a parametrized test.

Reply:
I'm not sure how a parameterized version of this test would work.  I can split these into individual tests (would should also be simpler given my above suggestion for `from_args()`).